interactive/dashboard/api.py

```diff
@@ -0,0 +1,1036 @@
+
+class DownloadConfig(views.APIView):
+    def get(self, request):
+        log("****** GET request received DOWNLOAD_CONFIG ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("session key: "+request.session.session_key)
+        destination_path = os.path.join(
+            settings.BASE_DIR,
+            "dashboard/static/zip/"
+            +request.session.session_key
+            +"/config_copy")
+        zip_path = os.path.join(
+            settings.BASE_DIR, "dashboard/static/zip/output/"
+            +request.session.session_key
+            +"/config.zip")
+        conf_path = os.path.join(
+            settings.BASE_DIR, 'configs/'+request.session.session_key)
+        log("destination path: "+destination_path)
+        log("zip path: "+zip_path)
+        log("conf path: "+conf_path)
+        create_config_zip(destination_path, zip_path, conf_path)
+
+        fl_path = zip_path
+        zip_file = open(fl_path, 'rb')
+        response = HttpResponse(zip_file, content_type='application/zip')
+        response['Content-Disposition'] = 'attachment; filename="%s"' % 'config.zip'
+        return response
+
+
+class DownloadResults(views.APIView):
+    def get(self, request):
+        log("****** GET request received DOWNLOAD_RESULTS ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("session key: "+request.session.session_key)
+        fl_path = os.path.join(
+            os.environ['MUSIC_BOX_BUILD_DIR'],
+            request.session.session_key+"/output.csv")
+        now = datetime.now()
+        filename = str(now) + '_model_output.csv'
+
+        fl = open(fl_path, 'r')
+        mime_type, _ = mimetypes.guess_type(fl_path)
+        response = HttpResponse(fl, content_type=mime_type)
+        response['Content-Disposition'] = "attachment; filename=%s" % filename
+        log("returning results:", response)
+        log("returning file:", filename)
+        return response
+
+
+class ConfigJsonUpload(views.APIView):
+    def post(self, request):
+        log("****** GET request received CONFIG_UPLOAD ******")
+        if not request.session.session_key:
+            request.session.create()
+        log("session key: "+request.session.session_key)
+        uploaded = request.FILES['file']
+        log("uploaded file: "+uploaded.name)
+        log("uploaded file size: "+str(uploaded.size))
+        configs_path = os.path.join(
+            settings.BASE_DIR, 'configs/'+request.session.session_key)
+        handle_uploaded_zip_config(
+            uploaded, "dashboard/static/zip/"+request.session.session_key,
+            configs_path)
+        export_to_path(configs_path+"/")
+        return HttpResponseRedirect('/mechanism.html')
+
+
+class RemoveInitialConditionsFile(views.APIView):
+    def get(self, request):
+        err = "removing initial conditions files should be a POST request"
+        return JsonResponse({"error":err})
```

Comment:
Seems like an appropriate place for a `BadRequest`

Reply:
But also should we even have this? The rest of our `POST`-only requests don't seem to have this kind of error checking for an incorrect HTTP verb. My vote is to remove this unless there is some other reason that I'm not aware of

Reply:
yea sure, after more testing it looks like django just immediately responds a bad request response so I'll remove these types of checks